File: 02-basic-challenges-2/01-fizzbuzz-array/fizzbuzz-array.js
function fizzBuzzArray(target) {
  const result = [];

  for (let num = 1; num <= target; num++) {
    if (num % 5 == 0 && num % 3 == 0) {
      result.push("FizzBuzz");
    } else if (num % 5 == 0) {
      result.push("Buzz");
    } else if (num % 3 == 0) {
      result.push("Fizz");
    } else {
      result.push(num);
    }
  }

  return result;
}

module.exports = fizzBuzzArray;

File: 02-basic-challenges-2/01-fizzbuzz-array/fizzbuzz-array.rb
def fizz_buzz_array(target)
  result = []

  (1..target).each do |num|
    result << if num % 5 == 0 && num % 3 == 0
      "FizzBuzz"
    elsif num % 5 == 0
      "Buzz"
    elsif num % 3 == 0
      "Fizz"
    else
      num
    end
  end

  result
end

puts(fizz_buzz_array(15))
